### PR TITLE
Add py-atlalign, the Atlas Alignement Python toolbox

### PR DESCRIPTION
**Pull request by pafonta**
## Description

Add what is needed for scientists and engineers to `use on BB5` the Python package `atlalign` ([PyPI](https://pypi.org/project/atlalign/)) .

## Validation

One could do `spack load py-atlalign` on BB5 to use the Python package atlalign on BB5.

## Dependencies

In order to package atlalign as a Spack package, additional Spack packaging was required.
All the `12 other package.py` are included in this pull request.

## Constraints

Not all version constraints were checked. Indeed, to do so, one should install `py-tensorflow` and `opencv +python`.
We discussed with @pramodk. These 2 Spack packages are difficult to install and I shouldn't try to install them.
Therefore, I don't know when submitting this PR if `spack install py-atlalign` would work.

## Packages

py-atlalign
+
py-antspyx
py-chart-studio
py-databricks-cli
py-gunicorn
py-imgaug
py-mlflow
py-nibabel
py-prometheus-flask-exporter
py-querystring-parser
py-tensorflow-addons
py-typeguard
py-webcolors

### CODE REVIEW

**Comment by pramodk** [comment]
Copying comment from Slack:

```
In order to test the PR and new recipes contributed, what if we comment out 
py-tensforlow dependency temporarily and add py-atlalign to deploy/environments/applications.yaml?
this way, all new recipes will be built under CI and we will find out if there are any issues.
```

@pafonta: On BB5, were you able to install these recipes? (excluding py-tensorflow)

**Comment by pafonta** [comment]
Hello @pramodk,

> @pafonta: On BB5, were you able to install these recipes? (excluding py-tensorflow)

It is `in progress`.

Strangely, some packages were installing without errors on a container with Ubuntu 20.04 but `not` on BB5.

I have committed the changes identified `so far` each time `spack install` on BB5 complains.

At the moment, I am on this error when `fontconfig-2.13.1` is installed at some point:

```
     169    checking for FREETYPE... no
  >> 170    configure: error: Package requirements (freetype2 >= 21.0.15) were not met:
     171    
     172    Package 'libpng', required by 'freetype2', not found
     173    
     174    Consider adjusting the PKG_CONFIG_PATH environment variable if you
     175    installed software in a non-standard prefix.
```

**Comment by pafonta** [comment]
By the way, is there a way to speed-up `spack install`?

Because, even on BBP, with `--fail-fast`, and without py-tensorflow, `spack install py-atlalign` takes forever...

**Comment by matz-e** [comment]
Retest this please.

Neurodamus testing seems broken, may require a rebase on latest `develop`.

How is the install slow? Are you starting from scratch every time? Do you have a log/configs you use?

**Comment by matz-e** [comment]
Yeah, looks like this requires a rebase.

**Comment by pafonta** [comment]
UPDATE Status

The branch `develop` has been merged in this PR, after #1204 was merged in `develop`, as agreed.

Tests are running.

File: var/spack/repos/builtin/packages/py-tensorflow-addons/package.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+class PyTensorflowAddons(PythonPackage):
+    """TensorFlow Addons."""
+
+    homepage = "https://pypi.org/project/tensorflow-addons/"
+    url = "https://github.com/tensorflow/addons/archive/refs/tags/v0.13.0.tar.gz"
+
+    version('0.13.0', sha256='5a8c33ecef5a3daca7f6b27c4b4e3a2badfe55f55a13e97066ee2e03ef98fab0')
+
+    depends_on('py-typeguard@2.7:', type=('run'))
```
Comment (pramodk) [inline var/spack/repos/builtin/packages/py-tensorflow-addons/package.py:19]: @pafonta : If I am not mistaken, this needs run dependency with `py-tensorflow` as well?

Comment (pafonta) [reply var/spack/repos/builtin/packages/py-tensorflow-addons/package.py:19]: Done in 729204aea9.

**Review by matz-e — COMMENTED** [review]
Following https://github.com/BlueBrain/spack/blob/develop/deploy/docs/deploy_bb5.md
please add `py-atlalign` to the `specs` key of
```
spack/deploy/environments/applications.yaml
```
and to the `whitelist` of
```
spack/deploy/configs/applications/modules.yaml
```

**Comment by pramodk** [comment]
@pafonta: FYI, As Matthias pointed out, we have to add `py-atalign` to `spack/deploy/environments/applications.yaml` and `spack/deploy/configs/applications/modules.yaml`. Otherwise the package is not installed and module is not created. See  f18298e.

**Comment by pramodk** [comment]
retest this please

**Comment by pramodk** [comment]
Unrelated to this PR but also [saw following](https://bbpcode.epfl.ch/ci/blue/organizations/jenkins/hpc.spack-deployment.pull-requests/detail/hpc.spack-deployment.pull-requests/2174/pipeline):

```

[2021-06-22T19:14:06.085Z] ==> Fetching https://spack-llnl-mirror.s3-us-west-2.amazonaws.com/_source-cache/archive/70/70172e58b6bad7572a3518c26729b074acdde15e6fee6cbab6d3528ad552b786.tar.gz
[2021-06-22T19:14:06.085Z] ==> py-pycairo: Executing phase: 'build'
[2021-06-22T19:14:11.460Z] ==> Fetching https://spack-llnl-mirror.s3-us-west-2.amazonaws.com/_source-cache/archive/66/6648a571a27f186e47094121f0095e1b809e918b3037c630c7f38ffad86e3035.tar.xz
[2021-06-22T19:14:11.460Z] ==> harfbuzz: Executing phase: 'autoreconf'
[2021-06-22T19:14:11.460Z] ==> harfbuzz: Executing phase: 'configure'
[2021-06-22T19:14:11.460Z] ==> Error: ValueError: Expecting value: line 1 column 1 (char 0)
[2021-06-22T19:14:11.460Z] 
[2021-06-22T19:14:11.460Z] /gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/pulls/1198/spack/var/spack/repos/builtin/packages/harfbuzz/package.py:48, in configure_args:
[2021-06-22T19:14:11.460Z]          45        # disable building of gtk-doc files following #9771
[2021-06-22T19:14:11.460Z]          46        args.append('--disable-gtk-doc-html')
[2021-06-22T19:14:11.460Z]          47        true = which('true')
[2021-06-22T19:14:11.460Z]   >>     48        args.append('CXXFLAGS={0}'.format(self.compiler.cxx11_flag))
[2021-06-22T19:14:11.460Z]          49        args.append('GTKDOC_CHECK={0}'.format(true))
[2021-06-22T19:14:11.460Z]          50        args.append('GTKDOC_CHECK_PATH={0}'.format(true))
[2021-06-22T19:14:11.460Z]          51        args.append('GTKDOC_MKPDF={0}'.format(true))
[2021-06-22T19:14:11.460Z] 
[2021-06-22T19:14:11.460Z] See build log for details:
[2021-06-22T19:14:11.460Z]   /gpfs/bbp.cscs.ch/ssd/slurmTmpFS/bbprelman/2266445/spack-stage-harfbuzz-2.6.8-t5oayoxngdm6arnmmu4bckwhan7snsor/spack-build-out.txt
[2021-06-22T19:14:11.460Z]
[2021-06-22T19:14:12.881Z] ==> Installing gobject-introspection-1.56.1-apvud4sdtyzkblmj4s3hsjaeje55lx62
[2021-06-22T19:14:12.881Z] ==> No binary for gobject-introspection-1.56.1-apvud4sdtyzkblmj4s3hsjaeje55lx62 found: installing from source
[2021-06-22T19:14:12.881Z] ==> Warning: Skipping build of pango-1.41.0-zkrytosqn4issv27lfnttrj36a57xfbe since harfbuzz-2.6.8-t5oayoxngdm6arnmmu4bckwhan7snsor failed
[2021-06-22T19:14:12.881Z] ==> Warning: Skipping build of py-xdot-1.0-7nonnlid62xglvnyotc4omseasnoljex since pango-1.41.0-zkrytosqn4issv27lfnttrj36a57xfbe failed
[2021-06-22T19:14:12.881Z] ==> Warning: Skipping build of stat-4.0.2.999-ggtrdgtzlkfw6lniltcus6wkbhf7djyg since py-xdot-1.0-7nonnlid62xglvnyotc4omseasnoljex failed
[2021-06-22T19:14:12.881Z] ==> Warning: Skipping build of graphviz-2.44.1-vzet323pzuuoeuapinuhnplag3rwz5si since pango-1.41.0-zkrytosqn4issv27lfnttrj36a57xfbe failed
[2021-06-22T19:14:12.881Z] ==> Warning: Skipping build of gtkplus-3.20.10-pckeucybkit6vgzdc3brzpecfznb4mou since pango-1.41.0-zkrytosqn4issv27lfnttrj36a57xfbe failed
[2021-06-22T19:14:12.881Z] ==> Warning: Skipping build of py-pygobject-3.38.0-sgw4gmxakkkifaw2krqa657vjfch3m6t since gtkplus-3.20.10-pckeucybkit6vgzdc3brzpecfznb4mou failed
[2021-06-22T19:14:12.881Z] ==> Warning: Skipping build of ghostscript-9.50-qayy6lkmj4ffygz3ctiywoiqi2rabnhi since gtkplus-3.20.10-pckeucybkit6vgzdc3brzpecfznb4mou failed
[2021-06-22T19:14:12.881Z] ==> Warning: Skipping build of groff-1.22.3-wp3nbjfdwuwr2yxvi2mkkxkgjeap4ms5 since ghostscript-9.50-qayy6lkmj4ffygz3ctiywoiqi2rabnhi failed
[2021-06-22T19:14:12.881Z] ==> Error: Failed to install harfbuzz due to InstallError: Install failed for dependency harfbuzz of stat
[2021-06-22T19:14:14.300Z] ==> py-pycairo: Executing phase: 'install'
[2021-06-22T19:14:16.259Z] [+] /gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/pulls/1198/deploy/externals/2021-06-22/linux-rhel7-x86_64/gcc-9.3.0/py-pycairo-1.18.1-3yckdd
[2021-06-22T19:14:20.537Z] ==> Fetching https://spack-llnl-mirror.s3-us-west-2.amazonaws.com/_source-cache/archive/5b/5b2875ccff99ff7baab63a34b67f8c920def240e178ff50add809e267d9ea24b.tar.xz
```

**Comment by pramodk** [comment]
irrelevant failure in last CI build:
```
[2021-06-22T23:54:36.005Z] sent 1,991 bytes  received 19 bytes  4,020.00 bytes/sec
[2021-06-22T23:54:36.005Z] total size is 60,537,187,969  speedup is 30,118,003.96
[2021-06-22T23:54:36.005Z] ### 01:54:35 starting deployment
[2021-06-22T23:54:36.005Z] ### 01:54:35 concretizing environment
[2021-06-23T00:41:05.639Z] wrapper script does not seem to be touching the log file in /gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/pulls/1198/spack/deploy@tmp/durable-bd051195
[2021-06-23T00:41:05.657Z] (JENKINS-48300: if on an extremely laggy filesystem, consider -Dorg.jenkinsci.plugins.durabletask.BourneShellScript.HEARTBEAT_CHECK_INTERVAL=86400)
script returned exit code -1
```
retest this please

File: var/spack/repos/builtin/packages/py-atlalign/package.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    depends_on('py-matplotlib@3.0.3:', type=('run'))
+    depends_on('py-mlflow', type=('run'))
+    depends_on('py-nibabel@2.4.0:', type=('run'))
+    depends_on('py-numpy', type=('run'))
+    depends_on('py-seaborn', type=('run'))
+    depends_on('py-scikit-image@0.16.0:', type=('run'))
+    depends_on('py-scikit-learn@0.20.2:', type=('run'))
+    depends_on('py-scipy', type=('run'))
+    depends_on('py-tensorflow@2.4.0:', type=('run'))
+    depends_on('py-tensorflow-addons', type=('run'))
```
Comment (lguyot) [inline var/spack/repos/builtin/packages/py-atlalign/package.py:31]: Hi Pierre-Alexandre, 

Here is the kind of test you could insert at the bottom of your package.py file (as discussed our last meeting with Matthias, Pramod and Francesco):

    depends_on('py-pytest', type='test')

```
    @run_after('install')
    @on_package_attributes(run_tests=True)
    def test_install(self):
        python("-m", "pytest", "tests/quick_test.py")
```

Comment (pafonta) [reply var/spack/repos/builtin/packages/py-atlalign/package.py:31]: Hello @lguyot,

Thanks but haven't @pramodk or @matz-e said that the integration test should be in some other file in some Gerrit repository instead?

Comment (lguyot) [reply var/spack/repos/builtin/packages/py-atlalign/package.py:31]: @pafonta You are certainly right. Let's see what they recommend.

**Comment by pafonta** [comment]
Thank you @pramodk and @matz-e for the last commits!

I hope the module will be able to build with the new modifications.

**Comment by matz-e** [comment]
Retest this please

**Review by matz-e — DISMISSED** [review]
Good from my side. @pramodk ?

**Review by pramodk — APPROVED** [review]
LGTM. As this has passed tests from @lguyot and @pafonta, this is good to go!

**Pull request merged**